docs/contributing/coding-guidelines/ros-nodes/console-logging.md

```diff
@@ -1,5 +1,128 @@
 # Console logging
 
-!!! warning
+ROS 2 logging is a powerful tool for understanding and debugging ros node.
 
-    Under Construction
+The logging subsystem in ROS 2 delivers logging messages to a variety of targets, including console, disk and ROS 2 network.
+
+This page focus on console logging and shows some useful example in practice. for detailed information, please see [logging documentation](https://docs.ros.org/en/humble/Concepts/About-Logging.html).
+
+## API
+
+There are four pattern of logging function:
+
+```bash
+RCLCPP_{severity}
+RCLCPP_{severity}_STREAM
+RCLCPP_{severity}_{condition}
+RCLCPP_{severity}_STREAM_{condition}
+```
+
+`STREAM` means output message in api is C++stream-style else printf style.
+
+`severity` have five choice:
+
+- `DEBUG, INFO, WARN, ERROR or FATAL`
+
+Set severity level to `DEBUG` only outputs message in debug mode.
+
+`condition` choices are:
+
+- `THROTTLE` - output the given printf-style message no more than the given rate
+- `ONCE` - output the given printf-style message only the first time this line is hit
+- `EXPRESSION` - output the given printf-style message only if the given expression is true
+- `FUNCTION` - output the given printf-style message only if the given function returns true
+- `SKIPFIRST` - output the given printf-style message all but the first time this line is hit
+
+## Console output formatting
+
+Default output format is [{severity}] [{time}] [{name}]: {message}
+
+To change log output format in console:
+
+```bash
+export RCUTILS_CONSOLE_OUTPUT_FORMAT="[{severity} {time}] [{name}]: {message} ({function_name}() at {file_name}:{line_number})"
+```
+
+- {severity} - The severity level.
+
+- {name} - The name of the logger (may be empty).
+
+- {message} - The log message (may be empty).
+
+- {function_name} - The function name this was called from (may be empty).
+
+- {file_name} - The file name this was called from (may be empty).
+
+- {time} - The time in seconds since the epoch.
+
+- {time_as_nanoseconds} - The time in nanoseconds since the epoch.
+
+- {line_number} - The line number this was called from (may be empty)
```

Comment:
It's already written in https://autowarefoundation.github.io/autoware-documentation/main/installation/additional-settings-for-developers/.

Could you update it if you feel it's unclear?

```suggestion
```